test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateE2ETests.cs

```diff
@@ -193,6 +223,34 @@ public void HandleResponse_Of_AnInvalidResponse_ThrowsException()
             Assert.Throws<DataServiceRequestException>(() => this.context.BulkUpdate<Person>(person));
         }
 
+        [Fact]
+        public async Task AsyncHandlingOf_AnInvalidResponse_ThrowsException()
+        {
+            var expectedResponse = "{\"@context\":\"http://localhost:8000/$metadata#Persons/$delta\",\"value\":[{\"ID\":100,\"Name\":\"Bing\",\"Cars@delta\":[{\"ID\":null,\"Name\":null}]}]}";
+
+            SetupContextWithRequestPipelineForSaving(
+                this.context,
+                expectedResponse);
+
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var car = new Car
+            {
+                ID = 1001,
+                Name = "A"
+            };
+
+            this.context.AttachTo("Persons", person);
+
+            this.context.AddRelatedObject(person, "Cars", car);
+
+            await Assert.ThrowsAsync<DataServiceRequestException>(async () => await this.context.BulkUpdateAsync<Person>(person));

```

Comment:
Is the exception expected to have any details about the error, error message? If so, I think we should also have assertions that verify that.

Reply:
I've added. The `DateServiceRequestException` has a `DataServiceResponse` property which will be empty whevever an exception is thrown. The exception will have a `Message` and an 'InnerException' but these will differ depending on the deserialization exception thrown.